src/main/java/com/redhat/devtools/lsp4ij/LSPIJUtils.java

```diff
@@ -595,12 +596,22 @@ public static int toOffset(int line, int character, @NotNull Document document)
 
         // Adjust position line/character according to this comment https://github.com/microsoft/vscode-languageserver-node/blob/ed3cd0f78c1495913bda7318ace2be7f968008af/textDocument/src/main.ts#L26
         if (line >= document.getLineCount()) {
-            // The line number is greater than the number of lines in a document, it defaults back to the number of lines in the document.
+            // The line number is greater than the number of lines in a document, it defaults to the last valid offset in the document.
             return document.getTextLength();
         } else if (line < 0) {
             // The line number is negative, it defaults to 0.
             return 0;
         }
+        // If the line number is 0 and the character is beyond the length of the first line in a multi-line document,
```

Comment:
Now the logic to detect an offset-only position and translate it to exactly the reported offset is _very_ specific. It will only occur if:
1. `line=0`
2. `character>lines[0].length`
3. The document has more than one line.
Again, it's technically out-of-spec, but I imagine that it should behave as-intended for all but significantly ill-behaved language servers.